datastream-server/src/main/java/com/linkedin/datastream/server/assignment/StickyPartitionAssignmentStrategy.java

```diff
@@ -87,6 +150,40 @@ public StickyPartitionAssignmentStrategy(Optional<Integer> maxTasks, Optional<In
 
     Validate.isTrue(totalTaskCount > 0, String.format("No tasks found for datastream group %s", dgName));
 
+    if (getEnableElasticTaskAssignment(datastreamPartitions.getDatastreamGroup()) && assignedPartitions.isEmpty()) {
```

Comment:
Can you extract a method for this code block?

Reply:
done